### PR TITLE
Fix arm docker build

### PR DESCRIPTION
**Pull request by mattcen**
History is at: https://matrix.to/#/!YTvKGNlinIzlkMTVRl:matrix.org/$4Se7Bpo2lakGZLGxnsXA8WQlCHtMLrJRs8qiNshzBKA?via=matrix.org&via=privacytools.io&via=mozilla.org
(Transcript below)

In short, trying to build an Arm binary of the Docker image for riot-web on either a Raspberry Pi 3 or a MacBook via Docker buildx, there are no pre-built binaries available for `canvas`, and building it causes `webpack` to run out of memory. These two commits seem to rectify these issues.

Chat transcript
---

Matt Cengia (he/him/his)
> Hey folks, I'm trying to get Riot installed, via Docker, onto a Raspberry Pi 3. When I try to build the Docker image, either on my RPi, or on an emulated Arm container on my Mac via docker buildx, I get an error about 'canvas'. I'm building v1.6.7, and when I try to build the same image as an amd64 image, it works fine. Any ideas what's going on? Here's a gist of the error I get (I tried manually stepping through the Docker container build to debug it):
> https://gist.github.com/cac6efa0e9730ae84397a6fad9a9609d

Michael (t3chguy)
> Matt Cengia (he/him/his): https://github.com/vector-im/riot-web/issues/14400#issuecomment-657106643

Matt Cengia (he/him/his)
> Ah, right. Thanks, Michael (t3chguy) . Is this something for which I could propose a patch to the Dockerfile?

Michael (t3chguy)
> Sure

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,7 +9,8 @@ ARG REACT_SDK_BRANCH="master"
 ARG JS_SDK_REPO="https://github.com/matrix-org/matrix-js-sdk.git"
 ARG JS_SDK_BRANCH="master"
 
-RUN apt-get update && apt-get install -y git dos2unix
+RUN apt-get update && apt-get install -y git dos2unix \
+  build-essential libcairo2-dev libpango1.0-dev libjpeg-dev libgif-dev librsvg2-dev
```
Comment (jryans) [inline Dockerfile:13]: Hmm, so why are these needed on Arm? Because of some native node module which not available pre-compiled...?

Comment (mattcen) [reply Dockerfile:13]: Yes. As per https://www.npmjs.com/package/canvas#compiling:
> If you don't have a supported OS or processor architecture, or you use --build-from-source, the module will be compiled on your system. This requires several dependencies, including Cairo and Pango.

So there's a pre-compiled binary available for amd64, but not for arm.

Comment (mattcen) [reply Dockerfile:13]: It's also worth noting, in case it's not obvious, that these are only being added to the *builder* image, not the final Docker image.

Comment (jryans) [reply Dockerfile:13]: Right, makes sense to me. Could you add a comment in the file to explain it's for compiling canvas on architectures like Arm? Then should be good to merge.

**Comment by jryans** [comment]
@mattcen Just a nudge to add a comment as mentioned above, and then we should be ready to merge this. Thanks for working on it!

**Comment by mattcen** [comment]
@jryans Ah, sorry, I missed your last comment about this; have just pushed another commit with a comment in it.
